app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import moment from 'moment'
+import _ from 'lodash'
+
+import SupplementalApplicationHeader from './SupplementalApplicationHeader'
+import SupplementalApplicationContainer from './SupplementalApplicationContainer'
+import TabsSection from '../organisms/TabsSection'
+import appPaths from '../../utils/appPaths'
+import mapProps from '../../utils/mapProps'
+
+const SupplementalApplicationPage = ({ application, statusHistory }) => {
+  // console.log(statusHistory)
+  const tabNames = [
+    { title: 'Short Form Application',    url: appPaths.toApplication(application.id) },
+    { title: 'Supplemental Information',  url: appPaths.toApplicationSupplementals(application.id) }
+  ]
+
+  const currentUrl = window.location.pathname
+
+  return (
+    <div>
+      <SupplementalApplicationHeader
+        applicationNumber={application.number}
+        applicantName={application.name}
+        listingId={application.listingId}/>
+      <TabsSection items={tabNames} currentUrl={currentUrl}>
+        <SupplementalApplicationContainer statusHistory={statusHistory} />
+      </TabsSection>
+    </div>
+  )
+}
+
+/*********************/
+/* Prop mapping      */
+/*********************/
+
+// I'm mapping properties passed to react_component to the actual Page Component,
+// so we do not propagate salesforce domain objects thru our components three.
+// Fed
+
+const mapApplicationProperty = (application) => {
+  return {
+    id: application.Id,
+    number: application.Name,
+    name: application.Applicant.Name,
+    listingId: application.Listing.Id
+  }
+}
+
+const mapStatusHistoryItem = (item) => {
+  return {
+    status: item.Processing_Status,
+    note: item.Processing_Comment,
+    date: item.Processing_Date_Updated,
+    timestamp: moment(item.Processing_Date_Updated).unix()
+  }
+}
+
+const mapStatusHistoryProperty = (statusHistory) => {
+  if (_.isEmpty(statusHistory))
+    return []
+  return statusHistory.map(mapStatusHistoryItem)
+}
+
+const mapProperties = ({application, statusHistory}) => {
+  return {
+    application: mapApplicationProperty(application),
+    statusHistory: mapStatusHistoryProperty(statusHistory)
+  }
+}
+
+export default mapProps(mapProperties)(SupplementalApplicationPage)
```

Comment:
My opinion is that ideally, this domain mapping would be a server-side concern and not pollute the front-end code. We have some ideas around that planned in this chore: https://www.pivotaltracker.com/story/show/154747783

However, it may be some time before we're able to get to that chore. For now, I think this front-end field mapping is a great idea, and lays a nice groundwork for the server side changes. When we do get to server-side changes, we can use the mapping logic here to inform how we restructure the Salesforce data, and then updates the front-end code will mostly be removing these mapping functions--making for a pretty clean transition.

So let's keep adding front-end mapping until we get to the server-side chore!

Reply:
@fddayan @creativecoder Yep as Grant says above this general idea is something we've talked about for a while, so overall the team is in agreement on mapping being a good + necessary thing. Seeing this, though, I'm wondering why we would spend time doing this in the frontend now and then re-doing this in the backend later. Is the work needed to do this server-side significantly greater than the work we see here? If the answer is yes, then what is here is good and let's roll with this. If the answer is no, then we should discuss bringing the server-side mapping story up in priority.

Reply:
We all discussed this in stand-up and agreed to approve this approach 👍

Reply:
this will get bigger and bigger, so as it gets bigger, we need to either pull it out to a central file now or later.